frontend/src/components/registration-form/registration.form.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useCallback, useState } from "react";
+import styles from "./registration.form.module.scss";
+import { AuthService } from "../../pages/api/authService";
+import { LongButton } from "shared-components";
+import { RegistrationConfirmation, RegistrationError, RegistrationFormFieldsets } from "./components";
+import { useRegistrationForm } from "./hooks/useRegistrationForm";
+import { scrollFirstFieldWithErrorsIntoView } from "./utils";
+import { RegistrationFormFieldsetsService } from "./registrationFormFieldsetsService";
+
+export const RegistrationForm = () => {
+	const registrationForm = useRegistrationForm();
+	const [shouldValidate, setShouldValidate] = useState(false);
+	const [hasServerError, setHasServerError] = useState(false);
+	const [successfullyRegisteredUserEmail, setSuccessfullyRegisteredUserEmail] = useState("");
+
+	const handleRegistrationError = useCallback((error) => {
+		if (error.response?.status === 400) {
+			registrationForm.setAPIValidationErrors(error.response.data.errors);
+			scrollFirstFieldWithErrorsIntoView();
+		} else {
+			setHasServerError(true);
+		}
+	}, [registrationForm]);
+
+	const register = useCallback(async () => {
+		const data = registrationForm.getAPIRequestData();
+		try {
+			await AuthService.register(data, registrationForm.profileType);
+			RegistrationFormFieldsetsService.deleteFieldsets();
+			setSuccessfullyRegisteredUserEmail(registrationForm.fieldsets.common.fields.email.value);
+		} catch (error) {
+			handleRegistrationError(error);
+		}
+	}, [handleRegistrationError, registrationForm]);
+
+	const submit = useCallback(async () => {
+		setShouldValidate(true);
+		if (registrationForm.hasErrors()) {
+			scrollFirstFieldWithErrorsIntoView();
+		} else {
+			await register();
+		}
+	}, [register, registrationForm]);
+
+	if (hasServerError) return <RegistrationError/>;
+
+	if (successfullyRegisteredUserEmail) return <RegistrationConfirmation email={successfullyRegisteredUserEmail}/>;
+
+	if (!registrationForm.fieldsets) return null;
+
+	return (
+		<div className={styles.background}>
+			<div className={styles.box}>
+				<div className={styles.boxHeader}>
+					<h4 className={styles.h4}>Napravi profil</h4>
+				</div>
+				<form>
+					<RegistrationFormFieldsets
+						profileType={registrationForm.profileType}
+						fieldsets={registrationForm.fieldsets}
+						onFieldUpdate={registrationForm.updateField}
+						shouldValidate={shouldValidate}
+					/>
+				</form>
+				{/*TODO: ADD NOT ROBOT */}

```

Comment:
Ovaj to do mozemo ukloniti kapiram?

Reply:
@milos1302 , vidim isto da nema vise ogranicenje za password za 8 karaktera. Mozda da vratimo ipak to ogranicenje?

Reply:
Ima to isto ogranicenje, samo sto su sad "karakteristike" field-a u field objektima, pa se na osnovu njih dodavaju validacije tom fieldu prilikom njihovog inicalizovanja.
Npr. ova dva propertija:
![image](https://user-images.githubusercontent.com/39025384/209684893-84b7a119-67c7-4fdf-bde2-5482e214bdca.png)
govore hook-u `useRegistrationForm` da prilikom "pripreme" field-ova doda ove dve validacije tom field-u:
![image](https://user-images.githubusercontent.com/39025384/209685077-de07815b-87e3-4efb-9879-ff8d0ed95cca.png)

Reply:
Sto se tice ovog `TODO: ADD NOT ROBOT`, mislim da je ok da to ostavimo u kodu s obzirm da je to i dalje jedan od TODO-ova koje bi bilo pozeljno napraviti 😅

Reply:
@milos1302 , aha, nisam skapirao posto sam uspeo da posaljem zahtev sa jednim karakterom lozinke i nije mi izbacio error. Ajde proveri samo kod tebe u lokalu je l ispravno radi. Javi ako radi da probam opet kod sebe, posto sad ne mogu da se setim koji sam tacno repro step napravio da prodjem tu validaciju 😅

Reply:
U pravu si. Nije mu dodao tu validaciju jer `password` field nije updao u ovaj `if`:
![image](https://user-images.githubusercontent.com/39025384/209778562-eb7020d5-ce87-49a7-9d09-1601ad4efdea.png)

Uklonio sam taj `if`, tako da ce se kod svih field-ova proveriti taj `minLength` i `maxLength` i dodati im se ti validatori ukoliko je potrebno.